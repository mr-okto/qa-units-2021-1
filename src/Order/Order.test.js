/* eslint-disable import/first */
jest.mock('../utils/getDate');

import React from 'react'
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import {getDate} from '../utils/getDate';
import Order from "./Order";
import {fakeOrders} from "../data/fakeOrders";

configure({adapter: new Adapter()});

describe('Order.js', () => {
  let order;
  beforeEach(() => {
    order = Object.assign({}, fakeOrders[0]);
    getDate.mockReturnValue("11 марта, чт, 2021 год");
  });
  afterEach(() => {
    jest.resetModules();
  });

  it('render no order is null', () => {
    const wrapper = shallow(<Order />);
    expect(wrapper.getElement()).toBeNull();
  });

  it('render empty order is null', () => {
    const wrapper = shallow(<Order order={{}}/>);
    expect(wrapper.getElement()).toBeNull();
  });

  it('render order without shop is null', () => {
    delete order.shop;
    const wrapper = shallow(<Order order={order}/>);
    expect(wrapper.getElement()).toBeNull();
  });

  it('render order without date is null', () => {
    delete order.date;
    const wrapper = shallow(<Order order={order}/>);
    expect(wrapper.getElement()).toBeNull();
  });

  it('render order without items snapshot', () => {
    delete order.items;
    const wrapper = shallow(<Order order={order}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render fake order snapshot', () => {
    const wrapper = shallow(<Order order={order}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('getDate is called', () => {
    const wrapper = shallow(<Order order={order}/>);
    expect(getDate).toHaveBeenCalled();
  });
});

