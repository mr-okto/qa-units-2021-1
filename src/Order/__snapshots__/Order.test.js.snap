// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js render fake order snapshot 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      11 марта, чт, 2021 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order.js render order without items snapshot 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      11 марта, чт, 2021 год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;
